utils/db.py

```diff
@@ -92,13 +93,10 @@ class Db(Mapping):
         tables, like the mapping interface or :py:meth:`values`.
 
     """
-    def __init__(self, hostname=None, credentials=None):
+    def __init__(self, hostname=None, credentials=None, db_port=None):
         self._table_cache = {}
-        if hostname is None:
-            self.hostname = store.current_appliance.db_address
-        else:
-            self.hostname = hostname
-
+        self.hostname = hostname or store.current_appliance.db_address
+        self.db_port = db_port or store.current_appliance.db_port
```

Comment:
Can you check the usage of the Db class? Maybe we don't need the current appliance port here because the Db is only invoked by the appliance class itself which passes the DB port. If that is the case, then I would put `or ports.DB` for sake of compatibility and let the appliance object pass the DB port itself..